### PR TITLE
Jorunals_using_jsonb_yarin

### PR DESCRIPTION
**Pull request by yarinTag**
Journal convert data column to jsonb

### CODE REVIEW

File: src/modules/shiftJournals/shiftJournal.controller.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { ShiftJournal } from './shiftJournal.schema';
+
+class ShiftJournalController implements IShiftJournalController {
+  constructor(private repository: IShiftJournalRepository) {}
+
+  public createShiftJournal = async (data: Shift): Promise<void> => {
+    const journalEntry = this.repository.create(data);
+    const validationResult = await validationEntity(ShiftJournal, journalEntry);
+
+    if (validationResult.success === false) {
```
Comment (yariv245) [inline src/modules/shiftJournals/shiftJournal.controller.ts]: remove the if, no need

File: src/modules/shiftJournals/shiftJournal.repository.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return this.repository.save(shiftJournal);
+  }
+
+  create(data: Shift): ShiftJournal {
+    return this.repository.create({ data });
+  }
+
+  async findAllBy(workCycleId: string): Promise<ShiftJournal[]> {
+    return await this.repository
+      .createQueryBuilder('shiftJournal')
```
Comment (yariv245) [inline src/modules/shiftJournals/shiftJournal.repository.ts]:
```suggestion
      .createQueryBuilder(ShiftJournal.name)
```

check if it works -> if not create new const and use it

File: src/modules/shifts/shift.service.ts [hunk 3]
```diff
@@ -31,7 +35,8 @@ export class ShiftService implements IShiftService {
     }
 
     const result = await this.repository.save(shift);
-
```
Comment (yariv245) [inline src/modules/shifts/shift.service.ts]: remove it.

use the https://typeorm.io/listeners-and-subscribers

every time the shift repo does save/update -> it supposed to call it automatically.
I don't want us to set this row manually every time we do save/update to Shift

Comment (yarinTag) [reply src/modules/shifts/shift.service.ts]: using subscriber

File: src/modules/shiftJournals/shiftJournal.repository.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  save(shiftJournal: ShiftJournal): Promise<ShiftJournal> {
+    return this.repository.save(shiftJournal);
+  }
+
+  create(data: Shift): ShiftJournal {
+    return this.repository.create({ data });
+  }
+
+  async findAllBy(id: string): Promise<ShiftJournal[]> {
```
Comment (yariv245) [inline src/modules/shiftJournals/shiftJournal.repository.ts]:
```suggestion
  async findAllByShiftId(id: string): Promise<ShiftJournal[]> {
```

**Review by yariv245 — APPROVED** [review]

**Pull request merged**